### ISSUE TITLE
Bug when spamming date picker from showDatePicker() in debug mode.

### ISSUE THREAD

**Issue by JesseHines0** [flutter/flutter#32628]
If you make many selections in the date dialog it will stop showing the dates, and won't let you select any more. The date picker will still allow you to scroll months, but erase everything but an "s", and won't let you select or see any other dates. Closing and reopening the dialog does not fix the problem. 
  It seems to do it consistently if you select a date in the date picker 38-40 times. The bug will still occur if the clicks are spread across opening/closing/OKing of the date picker.
  It seems to only occur if I run the app in debug configuration, in release it works fine.
Before...
<img src="https://user-images.githubusercontent.com/47368565/57631633-e081d180-756d-11e9-88da-c0c8f2931041.png" width="200"/>
After ...
<img src="https://user-images.githubusercontent.com/47368565/57631639-e5468580-756d-11e9-90d9-06218508f7bc.png" width="200"/>
Code to reproduce:
```dart
  Future<void> selectDate(BuildContext context) async {
    DateTime date = await showDatePicker(
        context: context,
        initialDate: DateTime(2019, 5, 13),
        firstDate: DateTime(2019, 1, 1),
        lastDate: DateTime(2019, 12, 31),
    );
    return date;
  }
```
I'm running the app on an Samsung Galaxy J3 prime, Android 7.0.
```
$ flutter devices
    SM J327T • 42009800ce606435 • android-arm • Android 7.0 (API 24)
```
```
$ flutter doctor
    Doctor summary (to see all details, run flutter doctor -v):
    [✓] Flutter (Channel stable, v1.5.4-hotfix.2, on Linux, locale en_US.UTF-8)
 
    [✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    [✓] Android Studio (version 3.3)
    [✓] VS Code (version 1.33.1)
    [✓] Connected device (1 available)

    • No issues found!
```
```
$ flutter -version
    Flutter 1.5.4-hotfix.2 • channel stable • https://github.com/flutter/flutter.git
    Framework • revision 7a4c33425d (2 weeks ago) • 2019-04-29 11:05:24 -0700
    Engine • revision 52c7a1e849
    Tools • Dart 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)
 ``` 
It will sometimes print this error, but not reliably, so it may not be related...
```
I/flutter (19781): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (19781): The following assertion was thrown while finalizing the widget tree:
I/flutter (19781): _MonthPickerState#de11d(ticker active) was disposed with an active Ticker.
I/flutter (19781): _MonthPickerState created a Ticker via its SingleTickerProviderStateMixin, but at the time dispose()
I/flutter (19781): was called on the mixin, that Ticker was still active. The Ticker must be disposed before calling
I/flutter (19781): super.dispose(). Tickers used by AnimationControllers should be disposed by calling dispose() on the
I/flutter (19781): AnimationController itself. Otherwise, the ticker will leak.
I/flutter (19781): The offending ticker was: Ticker(created by _MonthPickerState#de11d(lifecycle state: created))
I/flutter (19781): The stack trace when the Ticker was actually created was:
I/flutter (19781): #0      new Ticker.<anonymous closure> 
I/flutter (19781): #1      new Ticker 
I/flutter (19781): #2      __MonthPickerState&State&SingleTickerProviderStateMixin.createTicker 
I/flutter (19781): #3      new AnimationController 
I/flutter (19781): #4      _MonthPickerState.initState 
I/flutter (19781): #5      StatefulElement._firstBuild 
I/flutter (19781): #6      ComponentElement.mount 
I/flutter (19781): #7      Element.inflateWidget 
I/flutter (19781): #8      Element.updateChild 
I/flutter (19781): #9      SingleChildRenderObjectElement.mount 
I/flutter (19781): #10     Element.inflateWidget 
I/flutter (19781): #11     Element.updateChild 
I/flutter (19781): #12     ComponentElement.performRebuild 
I/flutter (19781): #13     Element.rebuild 
I/flutter (19781): #14     ComponentElement._firstBuild 
I/flutter (19781): #15     ComponentElement.mount 
I/flutter (19781): #16     ParentDataElement.mount 
I/flutter (19781): #17     Element.inflateWidget 
I/flutter (19781): #18     MultiChildRenderObjectElement.mount 
I/flutter (19781): #19     Element.inflateWidget 
I/flutter (19781): #20     Element.updateChild 
I/flutter (19781): #21     SingleChildRenderObjectElement.mount 
I/flutter (19781): #22     Element.inflateWidget 
I/flutter (19781): #23     Element.updateChild 
I/flutter (19781): #24     ComponentElement.performRebuild 
I/flutter (19781): #25     Element.rebuild 
I/flutter (19781): #26     ComponentElement._firstBuild 
I/flutter (19781): #27     ComponentElement.mount 
I/flutter (19781): #28     Element.inflateWidget 
I/flutter (19781): #29     MultiChildRenderObjectElement.mount 
I/flutter (19781): #30     Element.inflateWidget 
I/flutter (19781): #31     Element.updateChild 
I/flutter (19781): #32     SingleChildRenderObjectElement.mount 
I/flutter (19781): #33     Element.inflateWidget 
I/flutter (19781): #34     Element.updateChild 
I/flutter (19781): #35     _LayoutBuilderElement._layout.<anonymous closure> 
I/flutter (19781): #36     BuildOwner.buildScope 
I/flutter (19781): #37     _LayoutBuilderElement._layout 
I/flutter (19781): #38     RenderObject.invokeLayoutCallback.<anonymous closure> 
I/flutter (19781): #39     PipelineOwner._enableMutationsToDirtySubtrees 
I/flutter (19781): #40     RenderObject.invokeLayoutCallback 
I/flutter (19781): #41     _RenderLayoutBuilder.performLayout 
I/flutter (19781): #42     RenderObject.layout 
I/flutter (19781): #43     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #44     RenderObject.layout 
I/flutter (19781): #45     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #46     RenderObject.layout 
I/flutter (19781): #47     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #48     _RenderCustomClip.performLayout 
I/flutter (19781): #49     RenderObject.layout 
I/flutter (19781): #50     RenderConstrainedBox.performLayout 
I/flutter (19781): #51     RenderObject.layout 
I/flutter (19781): #52     RenderPositionedBox.performLayout 
I/flutter (19781): #53     RenderObject.layout 
I/flutter (19781): #54     RenderPadding.performLayout 
I/flutter (19781): #55     RenderObject.layout 
I/flutter (19781): #56     RenderPadding.performLayout 
I/flutter (19781): #57     RenderObject.layout 
I/flutter (19781): #58     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #59     RenderObject.layout 
I/flutter (19781): #60     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #61     RenderObject.layout 
I/flutter (19781): #62     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #63     RenderObject.layout 
I/flutter (19781): #64     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #65     RenderObject.layout 
I/flutter (19781): #66     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #67     RenderObject.layout 
I/flutter (19781): #68     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #69     RenderObject.layout 
I/flutter (19781): #70     _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout 
I/flutter (19781): #71     RenderOffstage.performLayout 
I/flutter (19781): #72     RenderObject.layout 
I/flutter (19781): #73     RenderStack.performLayout 
I/flutter (19781): #74     RenderObject._layoutWithoutResize 
I/flutter (19781): #75     PipelineOwner.flushLayout 
I/flutter (19781): #76     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame 
I/flutter (19781): #77     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame 
I/flutter (19781): #78     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback 
I/flutter (19781): #79     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback 
I/flutter (19781): #80     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame 
I/flutter (19781): #81     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame 
I/flutter (19781): #85     _invoke  (dart:ui/hooks.dart:219:10)
I/flutter (19781): #86     _drawFrame  (dart:ui/hooks.dart:178:3)
I/flutter (19781): (elided 3 frames from package dart:async)
I/flutter (19781):
I/flutter (19781): When the exception was thrown, this was the stack:
I/flutter (19781): #0      __MonthPickerState&State&SingleTickerProviderStateMixin.dispose.<anonymous closure> 
I/flutter (19781): #1      __MonthPickerState&State&SingleTickerProviderStateMixin.dispose 
I/flutter (19781): #2      _MonthPickerState.dispose 
I/flutter (19781): #3      StatefulElement.unmount 
I/flutter (19781): #4      _InactiveElements._unmount 
I/flutter (19781): #5      _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #6      SingleChildRenderObjectElement.visitChildren 
I/flutter (19781): #7      _InactiveElements._unmount 
I/flutter (19781): #8      _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #9      ComponentElement.visitChildren 
I/flutter (19781): #10     _InactiveElements._unmount 
I/flutter (19781): #11     _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #12     MultiChildRenderObjectElement.visitChildren 
I/flutter (19781): #13     _InactiveElements._unmount 
I/flutter (19781): #14     _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #15     SingleChildRenderObjectElement.visitChildren 
I/flutter (19781): #16     _InactiveElements._unmount 
I/flutter (19781): #17     _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #18     ComponentElement.visitChildren 
I/flutter (19781): #19     _InactiveElements._unmount 
I/flutter (19781): #20     _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #21     MultiChildRenderObjectElement.visitChildren 
I/flutter (19781): #22     _InactiveElements._unmount 
I/flutter (19781): #23     _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #24     SingleChildRenderObjectElement.visitChildren 
I/flutter (19781): #25     _InactiveElements._unmount 
I/flutter (19781): #26     _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #27     _LayoutBuilderElement.visitChildren 
I/flutter (19781): #28     _InactiveElements._unmount 
I/flutter (19781): #29     _InactiveElements._unmount.<anonymous closure> 
I/flutter (19781): #30     ComponentElement.visitChildren
```

**Comment by darrenaustin** [comment]
I have been unable to reproduce this using the sample code and instructions above. Not sure what would be going on here. The error reported does indeed point to a real problem that I fixed in #32843, but it is not clear if that was causing your problem. I am going to close this out for now, if you can still reproduce this with the above fix, please reopen and we can take another look.

**Comment by fluttercorner** [comment]
[How To Use Date Picker in Flutter](https://fluttercorner.com/how-to-use-date-picker-in-flutter/)
